service/src/event/service.rs

```diff
@@ -139,20 +139,16 @@ impl CeramicEventService {
         Ok((EventInsertable::try_new(event_id, body)?, metadata))
     }
 
-    pub(crate) async fn insert_events<'a>(
+    pub(crate) async fn insert_events(
         &self,
-        items: &[recon::ReconItem<'a, EventId>],
+        items: &[recon::ReconItem<EventId>],
         source: DeliverableRequirement,
     ) -> Result<InsertResult> {
-        if items.is_empty() {
-            return Ok(InsertResult::default());
-        }
-
-        let mut to_insert = Vec::with_capacity(items.len());
+        let mut to_insert = Vec::new();
 
         for event in items {
             let insertable =
-                Self::validate_discovered_event(event.key.to_owned(), event.value).await?;
+                Self::validate_discovered_event(event.key.to_owned(), &event.value).await?;
```

Comment:
is it correct that this function is where you see the main validation logic going?

Reply:
yeah, something like this. I was thinking we might want to change it to be a `self.validate_events(&[ReconItem<EventID>])` or something.. it could split things into unsigned, init that are valid, data, and time (anything unparsed is returned as invaild). if init is missing for data, it goes back as "pending", time events can be given to another task to process, and then we can `try_join!` (or something) on the set of init/data events and time events and return the result for every key (the #473 changes give a better store interface to actually do this).